### PR TITLE
CTCP-3663: Migration from Hardcoded codelists to Reference data

### PR DESCRIPTION
**Pull request by kevincoatsworth**

### CODE REVIEW

File: app/services/MeansOfTransportIdentificationTypesActiveService.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  private def filter(
+    identificationTypes: Seq[Identification],
+    index: Index,
+    borderModeOfTransport: Option[BorderModeOfTransport]
+  ): Seq[Identification] =
+    if (index.isFirst) {
+      borderModeOfTransport match {
+        case Some(borderMode) =>
+          identificationTypes.filterNot(_.code == UnknownIdentificationActive).filter(_.code.startsWith(borderMode.borderModeType.toString))
+        case _ => identificationTypes.filterNot(_.code == UnknownIdentificationActive)
```
Comment (joecefnhedog) [inline app/services/MeansOfTransportIdentificationTypesActiveService.scala:45]: we use `_.code == UnknownIdentificationActive)` twice, we could pull it out to make it DRYer

File: app/services/MeansOfTransportIdentificationTypesActiveService.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  private def filter(
+    identificationTypes: Seq[Identification],
+    index: Index,
+    borderModeOfTransport: Option[BorderModeOfTransport]
+  ): Seq[Identification] =
+    if (index.isFirst) {
+      borderModeOfTransport match {
+        case Some(borderMode) =>
+          identificationTypes.filterNot(_.code == UnknownIdentificationActive).filter(_.code.startsWith(borderMode.borderModeType.toString))
```
Comment (joecefnhedog) [inline app/services/MeansOfTransportIdentificationTypesActiveService.scala:44]: Quite hard to read this, can we refactor or maybe pull some of it into a sub function

**Review by joecefnhedog — COMMENTED** [review]
Looks good

File: app/services/MeansOfTransportIdentificationTypesActiveService.scala [hunk 3]
```diff
@@ -37,16 +37,19 @@ class MeansOfTransportIdentificationTypesActiveService @Inject() (referenceDataC
...
       borderModeOfTransport match {
         case Some(borderMode) =>
-          identificationTypes.filterNot(_.code == UnknownIdentificationActive).filter(_.code.startsWith(borderMode.borderModeType.toString))
-        case _ => identificationTypes.filterNot(_.code == UnknownIdentificationActive)
+          identificationTypesExcludingUnknown.filter(_.code.startsWith(borderMode.borderModeType.toString))
+        case _ => identificationTypesExcludingUnknown
       }
     } else {
-      identificationTypes.filterNot(_.code == UnknownIdentificationActive)
+      identificationTypesExcludingUnknown
```
Comment (joecefnhedog) [inline app/services/MeansOfTransportIdentificationTypesActiveService.scala:50]: This looks good, I think we don't need the outer if anymore

**Review by joecefnhedog — APPROVED** [review]

**Review by joecefnhedog — APPROVED** [review]

**Review by joecefnhedog — DISMISSED** [review]

**Review by joecefnhedog — DISMISSED** [review]

**Review by jackheslop96 — APPROVED** [review]

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**